src/mono/wasm/runtime/startup.ts

```diff
@@ -36,18 +36,33 @@ export function configure_emscripten_startup(module: DotnetModule, exportedAPI:
         (typeof (globalThis.document) === "object") &&
         (typeof (globalThis.document.createElement) === "function")
     ) {
+        module.baseUrl = module.baseUrl || "./";
+        module.mainScriptPath = module.mainScriptPath || "dotnet.js";
+
         // blazor injects a module preload link element for dotnet.[version].[sha].js
         const blazorDotNetJS = Array.from(document.head.getElementsByTagName("link")).filter(elt => elt.rel !== undefined && elt.rel == "modulepreload" && elt.href !== undefined && elt.href.indexOf("dotnet") != -1 && elt.href.indexOf(".js") != -1);
         if (blazorDotNetJS.length == 1) {
-            const hr = blazorDotNetJS[0].href;
-            console.log("determined url of main script to be " + hr);
+            const hr = blazorDotNetJS[0].href; 
+
+            var slashIndex = hr.lastIndexOf('/');
+
+            if(slashIndex != -1){
+                var urlPathName = new URL(hr).pathname;
+
+                module.mainScriptPath = urlPathName.substring(slashIndex+1);
+                module.baseUrl = urlPathName.substring(0, slashIndex);                
+            } else {
+                module.mainScriptPath = hr;
+            }
+
             (<any>module)["mainScriptUrlOrBlob"] = hr;
         } else {
```

Comment:
while we're changing this, it might make sense to have a convention for how to tell the runtime what its baseUrl is so it doesn't have to guess. i'm guessing uno could pipe that through

Reply:
The guessing should be removed - and blazor updated to provide a base URL (and possibly also the URL of the worker script - actually depending on how much stuff they mangle we might need some locate function where we can ask for normal names that we know and blazor can tell us what they mangled them to).  the guessing was just a hack because I didn't want to coordinate a pair of aspnetcore and runtime PRs